src/comment.rs

```diff
@@ -446,36 +448,47 @@ struct ItemizedBlock {
-    fn new(line: &str) -> ItemizedBlock {
-        let space_to_sigil = line.chars().take_while(|c| c.is_whitespace()).count();
-        // +2 = '* ', which will add the appropriate amount of whitespace to keep itemized
-        // content formatted correctly.
-        let mut indent = space_to_sigil + 2;
-        let mut line_start = " ".repeat(indent);
-
-        // Markdown blockquote start with a "> "
-        if line.trim_start().starts_with(">") {
-            // remove the original +2 indent because there might be multiple nested block quotes
-            // and it's easier to reason about the final indent by just taking the length
-            // of th new line_start. We update the indent because it effects the max width
-            // of each formatted line.
-            line_start = itemized_block_quote_start(line, line_start, 2);
-            indent = line_start.len();
+    /// Returns the sigil's (e.g. "- ", "* ", or "1. ") length or None if there is no sigil.
+    fn get_sigil_length(trimmed: &str) -> Option<usize> {
+        if trimmed.starts_with("* ") || trimmed.starts_with("- ") || trimmed.starts_with("> ") {
+            return Some(2);
         }
-        ItemizedBlock {
-            lines: vec![line[indent..].to_string()],
-            indent,
-            opener: line[..indent].to_string(),
-            line_start,
+
+        for suffix in [". ", ") "] {
+            if let Some((prefix, _)) = trimmed.split_once(suffix) {
+                if prefix.len() <= 2 && prefix.chars().all(|c| char::is_ascii_digit(&c)) {
+                    return Some(prefix.len() + suffix.len());
+                }
+            }
         }
+
+        None
+    }
+
+    /// Creates a new ItemizedBlock described with the given `line`
+    /// or None if `line` doesn't start an item.
+    fn new(line: &str) -> Option<ItemizedBlock> {
+        ItemizedBlock::get_sigil_length(line.trim_start()).map(|sigil_length| {
+            let space_to_sigil = line.chars().take_while(|c| c.is_whitespace()).count();
+            let mut indent = space_to_sigil + sigil_length;
+            let mut line_start = " ".repeat(indent);
+
+            // Markdown blockquote start with a "> "
+            if line.trim_start().starts_with(">") {
+                // remove the original +2 indent because there might be multiple nested block quotes
+                // and it's easier to reason about the final indent by just taking the length
+                // of the new line_start. We update the indent because it effects the max width
+                // of each formatted line.
+                line_start = itemized_block_quote_start(line, line_start, 2);
+                indent = line_start.len();
+            }
+            ItemizedBlock {
+                lines: vec![line[indent..].to_string()],
+                indent,
+                opener: line[..indent].to_string(),
+                line_start,
+            }
+        })
```

Comment:
Considering that `new` now returns an `Option` couldn't we remove a level of indentation by using the try (`?`) operator.

```suggestion
        let sigil_length = ItemizedBlock::get_sigil_length(line.trim_start())?;
        let space_to_sigil = line.chars().take_while(|c| c.is_whitespace()).count();
        let mut indent = space_to_sigil + sigil_length;
        let mut line_start = " ".repeat(indent);

        // Markdown blockquote start with a "> "
        if line.trim_start().starts_with(">") {
            // remove the original sigil_length because there might be multiple nested block quotes
            // and it's easier to reason about the final indent by just taking the length
            // of the new line_start. We update the indent because it effects the max width
            // of each formatted line.
            line_start = itemized_block_quote_start(line, line_start, sigil_length);
            indent = line_start.len();
        }
        ItemizedBlock {
            lines: vec![line[indent..].to_string()],
            indent,
            opener: line[..indent].to_string(),
            line_start,
        }
    })
```

Reply:
Good idea - thanks!  Done.